### ISSUE TITLE
bug: no level one heading in apps

### ISSUE THREAD

**Issue by darrylnoakes** [ionic-team/ionic-framework#23746]
### Prequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already include this feature request, without success.

### Describe the Feature Request

Make the internal element used by `ion-title` an `h1` when used in a page header.
This improves accessibility.

### Describe the Use Case

This greatly improves accessibility.
There are many more small changes like this that could be made throughout the framework...

### Describe Preferred Solution

`ion-title` currently always uses a `div`. An actual header element could be used instead.

**Comment by liamdebeasi** [comment]
Thanks for the issue. Is there a particular issue you are running into with the way `ion-title` is structured right now? `ion-title` can be used for more than just top level headings, so I think forcing it to use `h1` will cause additional accessibility issues.

**Comment by DarrylNoakes** [comment]
I did state that it should be "when used in a page header", but maybe I didn't make my request clear enough.
I was wondering if the the `ion-title` could detect if it was _naming a page_ (i.e. in a page header), and use an `h1` instead of a `div` only then.

I understand if this would be too complex. If that is the case, please say so, and I will (try to) forget about it.

Alternative, rather hacky solution I think should work for most cases: just put an `h*` into the slot.
`<ion-title><h1>Title of the Page</h1></ion-title>  `

**Comment by liamdebeasi** [comment]
Thanks for the clarification. I worry that adding this logic into `ion-title` would rely on it having to make (possibly incorrect) assumptions about how a dev's app is laid out.

Are you running into a particular accessibility issue/warning?

**Comment by DarrylNoakes** [comment]
Currently, axe DevTools gives a warning that the page has no level-one heading, and Windows Narrator (I have not tried other screen readers yet) does not read the title out.

It reads the back button I have in the toolbar, and then reads the content of the page.

**Comment by liamdebeasi** [comment]
Thanks. I think this is something Ionic should handle/make easier. One idea is to keep the scope extremely limited and just have it render an `h1` if it is used inside of `ion-header`.

**Comment by liamdebeasi** [comment]
Thanks for the issue. After making a [proof of concept](https://github.com/ionic-team/ionic-framework/pull/28710) and discussing this with the team we have decided to [add documentation ](https://github.com/ionic-team/ionic-docs/pull/3585) to inform devs on how to add proper headings in their applications.

We found that trying to automatically account for this in Ionic resulted in very brittle code because `ion-title` can be use in multiple ways (including using multiple titles within a single header).

The docs are live, so I am going to close this. Let me know if you have any questions.
